classes/booking_manager.php

```diff
@@ -273,6 +290,35 @@ public function validate($timenow = null): array {
                     new lang_string('error:invalidstatusspecified', 'mod_facetoface', $entry->status),
                 ];
             }
+
+            if (isset($userid)) {
+                if (!isset($userrows[$userid])) {
+                    $userrows[$userid] = [];
+                }
+                if (!isset($usersessions[$userid])) {
+                    $usersessions[$userid] = [];
+                }
+
+                $userrows[$userid][$row] = $entry->email;
+
+                if ($session && !in_array($session->id, $usersessions[$userid])) {
+                    $usersessions[$userid][] = $session->id;
+                }
+            }
```

Comment:
The reason I didn't do a simple count is because multiple rows for the same session is technically valid, so we need to know which distinct sessions the user is being added to. We also need the details of the email and rows so that we can create the errors. I have reworked this though so it's more like the session over capacity validation.

Reply:
Yep that makes sense :+1: